src/classlibnative/bcltype/system.cpp

```diff
@@ -423,6 +423,15 @@ void SystemNative::GenericFailFast(STRINGREF refMesgString, EXCEPTIONREF refExce
     WCHAR  *pszMessage = NULL;
     DWORD   cchMessage = (gc.refMesgString == NULL) ? 0 : gc.refMesgString->GetStringLength();
 
+    WCHAR * errorSourceString = NULL;
+
+    if (errorSource != NULL) {
+        DWORD cchErrorSource = errorSource->GetStringLength();
+        errorSourceString = new (nothrow) WCHAR[cchErrorSource + 1];
```

Comment:
`(nothrow)` means that this can return null. You should probably just ignore the passed in string when that happens.